warmup/summary_statistics.py

```diff
@@ -0,0 +1,299 @@
+                                                 key=lambda x: x[1], reverse=True))
+                cat_counts = list()
+                for category in cats_sorted:
+                    if cats_sorted[category] == 0:
+                        continue
+                    cat_counts.append('$%d$%s' % (cats_sorted[category], STYLE_SYMBOLS[category]))
+                reported_category += ' \\scriptsize(%s)' % ', '.join(cat_counts)
+            elif (sum(bmark['detailed_classification'].values()) ==
+                  bmark['detailed_classification'][bmark['classification']]):
+                # Consistent benchmark with no errors.
+                reported_category = STYLE_SYMBOLS[bmark['classification']]
+            else:  # No inconsistencies, but some process executions errored.
+                reported_category = ' %s\\scriptsize{($%d$)}' % \
+                                    (STYLE_SYMBOLS[bmark['classification']],
+                                     bmark['detailed_classification'][bmark['classification']])
+            if bmark['steady_state_iteration'] is not None:
+                mean_steady_iter = format_median_error(bmark['steady_state_iteration'],
+                                                       bmark['steady_state_iteration_confidence_interval'],
+                                                       as_integer=True)
+            else:
+                mean_steady_iter = ''
+            if bmark['steady_state_time'] is not None:
+                mean_steady = format_median_error(bmark['steady_state_time'],
+                                                  bmark['steady_state_time_confidence_interval'])
+            else:
+                mean_steady = ''
+            if bmark['steady_state_time_to_reach_secs'] is not None:
+                time_to_steady = format_median_error(bmark['steady_state_time_to_reach_secs'],
+                                                     bmark['steady_state_time_to_reach_secs_confidence_interval'],
+                                                     brief=True)
+            else:
+                time_to_steady = ''
+            latex_summary[vm][bmark['benchmark_name']] = {'style': reported_category,
+                'last_cpt': mean_steady_iter, 'last_mean': mean_steady,
+                'time_to_steady_state':time_to_steady}
+    return machine, list(sorted(benchmark_names)), latex_summary
+
+
+def write_latex_table(machine, all_benchs, summary, tex_file, num_splits, with_preamble=False):
+    """Write a tex table to disk"""
+
+    num_benchmarks = len(all_benchs)
+    all_vms = sorted(summary.keys())
+    num_vms = len(summary)
+
+    # decide how to lay out the splits
+    num_vms_rounded = \
+        int(math.ceil(num_vms / float(num_splits)) * num_splits)
+    vms_per_split = int(num_vms_rounded / float(num_splits))
+    splits = [[] for x in xrange(num_splits)]
+    vm_num = 0
+    split_idx = 0
+    for vm_idx in xrange(num_vms_rounded):
+        try:
```

Comment:
But `vm_idx` is a number and `all_vms` is a list (right? or maybe I got that bit wrong!), so the try/catch is just confusing AFAICT.